xbmc/music/tags/TagLoaderTagLib.cpp

```diff
@@ -462,6 +464,45 @@ bool CTagLoaderTagLib::ParseTag(ID3v2::Tag *id3v2, EmbeddedArt *art, MUSIC_INFO:
           tag.SetUserrating(POPMtoXBMC(popFrame->rating()));
         }
       }
+    else if (it->first == "CHAP")
+    {
+      // We use a chapter number index here to ensure chapters are added in the right order
+      int chapNumber = 0;
+      unsigned int prevTime = 0;
+      for (ID3v2::FrameList::ConstIterator ch = it->second.begin(); ch != it->second.end(); ++ch)
+      {
+        auto* chapFrame = dynamic_cast<ID3v2::ChapterFrame*>(*ch);
+        if (!chapFrame)
+          continue;
+        const std::string chapData = chapFrame->toString().toCString(true);
+        const std::vector<std::string> chapInfo = StringUtils::Split(chapData, ",");
+        if (!chapFrame->embeddedFrameList("TIT2").isEmpty())
+          chapters[chapNumber].chapterName =
+              chapFrame->embeddedFrameList("TIT2")[0]->toString().toCString(true);
+        const unsigned int startTime = chapFrame->startTime(); // times are in millisecs
+        unsigned int endTime = chapFrame->endTime();
+        std::vector<std::string> titles;
+        titles = StringUtils::Split(chapInfo[0], ":");
+        if (chapters[chapNumber].chapterName.empty())
```

Comment:
A small optimization suggestion: Instead of executing the code for `chapters[chapNumber]`several times, you could use a non const reference to hold that value.

```
ChapterDetails& details{chapters[chapNumber]};
```

This will either return an existing member if the map or create it.

Reply:
Thanks!  Done